lib/loader/loader.js

```diff
@@ -217,6 +217,11 @@ var loader, define, requireModule, require, requirejs;
   };
 
   define = function(name, deps, callback) {
+    var module = registry[name];
+    if (module && module.state !== 'new' && module.state !== 'errored') {
```

Comment:
> With the `errored` state, the module has already gone reification and its deps have caused an exception state to occur which is why the parent module is put in `errored` state.

This isn't actually accurate, the module doesn't become `reified` until _after_ all it's dependencies have been evaluated. Thus, if an error occurs in the dependencies, reification never happens. I confirmed this in the test for the `errored` state ([see here](https://github.com/ember-cli/loader.js/pull/104/files#diff-d0919eb0b787049e11a3548e180074fdR1549)).